src/libs/ascent/runtimes/flow_filters/ascent_runtime_rover_filters.cpp

```diff
@@ -242,62 +252,35 @@ RoverXRay::execute()
   }
   std::shared_ptr<VTKHCollection> collection = data_object->as_vtkh_collection();
 
-  std::string absorption = params()["absorption"].as_string();
+  // Validate that the 'absorption' field exists in the dataset
+  const std::string absorption = params()["rover/absorption"].as_string();
   if(!collection->has_field(absorption))
   {
     ASCENT_ERROR("Absorption field name '" << absorption << "' is not in the dataset");
   }
 
+  // Fetch the dataset associated with the 'absorption' field
   std::string topo_name = collection->field_topology(absorption);
-
-  // Returns an empty dataset if topo_name doesn't exist in the collection
+  // TODO: Validate topo_name
   vtkh::DataSet &dataset = collection->dataset_by_topology(topo_name);
```

Comment:
Can we also add a note to validate the `dataset`? We should figure out what `dataset_by_topology` returns if it gets a bad topo name. I would recommend reading docs or finding the source code.